src/UI/components/TodoButton.vue

```diff
@@ -1,5 +1,5 @@
 <template>
-  <button>{{ buttonName }}</button>
+  <button @click="emitValue">{{ buttonName }}</button>
```

Comment:
add onClick event on the component where it is being called so that you wont have to emit a function inside button component